moj-be/modules/custom/moj_resources/src/RelatedContentApiClass.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Drupal\moj_resources;
+
+use Drupal\node\NodeInterface;
+use Drupal\Core\Entity\Query\QueryFactory;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+/**
+ * RelatedContentApiClass
+ */
+
+class RelatedContentApiClass
+{
+    /**
+     * Node IDs
+     *
+     * @var array
+     */
+    protected $nids = array();
+    /**
+     * Nodes
+     *
+     * @var array
+     */
+    protected $nodes = array();
+    /**
+     * Language Tag
+     *
+     * @var string
+     */
+    protected $lang;
+    /**
+     * Node_storage object
+     *
+     * @var Drupal\Core\Entity\EntityManagerInterface
+     */
+    protected $node_storage;
+    /**
+     * Entitity Query object
+     *
+     * @var Drupal\Core\Entity\Query\QueryFactory
+     * 
+     * Instance of querfactory 
+     */
+    protected $entity_query;
+    /**
+     * Class Constructor 
+     *
+     * @param EntityTypeManagerInterface $entityTypeManager
+     * @param QueryFactory $entityQuery
+     */
+    public function __construct(
+        EntityTypeManagerInterface $entityTypeManager, 
+        QueryFactory $entityQuery
+    ) {
+        $this->node_storage = $entityTypeManager->getStorage('node');
+        $this->entity_query = $entityQuery;
+    }
+    /**
+     * API resource function
+     *
+     * @param [string] $lang
+     * @return array
+     */
+    public function RelatedContentApiEndpoint($lang, $category)
+    {
+        $this->lang = $lang;
+        $this->nids = self::getRelatedContentNodeIds($category);
+        $this->nodes = self::loadNodesDetails($this->nids);
+        // usort($this->nodes, 'self::sortByWeight');
+        return array_map('self::translateNode', $this->nodes);
+    }
+    /**
+     * TranslateNode function
+     *
+     * @param NodeInterface $node
+     * 
+     * @return $node
+     */
+    protected function translateNode(NodeInterface $node) 
+    {
+        return $node->hasTranslation($this->lang) ? $node->getTranslation($this->lang) : $node;
+    }
+    /**
+     * Get nids
+     *
+     * @return void
+     */
+    protected function getRelatedContentNodeIds($category)
+    {
+        $results = $this->entity_query->get('node')
+            ->condition('status', 1)
+            ->accessCheck(false);
+
+        if ($category !== 0) {
+            $results->condition('field_moj_top_level_categories', $category);
+        };
+        return $results->execute();
+    }
+    /**
+     * Load full node details
+     *
+     * @param array $nids
+     * @return array 
+     */
+    protected function loadNodesDetails(array $nids)
+    {
+        return array_filter(
+            $this->node_storage->loadMultiple($nids), function ($item) 
+            {
+                return $item->access();
+            }
+        );
+    }
+    /**
+     * sortByWeight
+     *
+     */
+    protected function sortByWeight($a, $b)
```

Comment:
Maybe you should name this based on the sorting `descending` or `ascending`?

Reply:
Made the change